src/pEpEngine.c

```diff
@@ -4176,11 +4177,277 @@ PEP_STATUS set_all_userids_to_own(PEP_SESSION session, identity_list* id_list) {
     return status;    
 }
 
-
 /* Temporary compatibility definitions
  * ***************************************************************** */
 DYNAMIC_API void set_debug_color(PEP_SESSION session, int ansi_color)
 {
     PEP_REQUIRE_ORELSE(session != NULL, { return; });
     LOG_WARNING("deprecated function");
 }
+
+static PEP_STATUS
+own_identity_by_address(PEP_SESSION session, const char *address, pEp_identity **identity)
+{
+    PEP_REQUIRE(address);
+    PEP_REQUIRE(identity);
+
+    if (EMPTYSTR(address)) {
+        return PEP_ILLEGAL_VALUE;
+    }
+
+    *identity = NULL;
+
+    identity_list *own_identities = NULL;
+    pEp_identity *found_identity = NULL;
+
+    PEP_STATUS status = own_identities_retrieve(session, &own_identities);
+    if (status != PEP_STATUS_OK) {
+        return status;
+    }
+
+    for (identity_list *identities = own_identities; identities; identities = identities->next) {
+        pEp_identity *identity = identities->ident;
+
+        if (!identity) {
+            continue;
+        }
+
+        if (EMPTYSTR(identity->address)) {
+            continue;
+        }
+
+        if (EMPTYSTR(identity->fpr)) {
+            continue;
+        }
+
+        if (!strcmp(identity->address, address)) {
+            found_identity = identity_dup(identity);
+            break;
+        }
+    }
+
+    free_identity_list(own_identities);
+
+    if (found_identity) {
+        *identity = found_identity;
+        return PEP_STATUS_OK;
+    } else {
+        return PEP_CANNOT_FIND_IDENTITY;
+    }
+}
+
+static PEP_STATUS
+sign_something(PEP_SESSION session, const pEp_identity *identity)
+{
+    const char *data = "DATA";
+    const size_t data_size = strlen(data) - 1;
+    char *signed_data = NULL;
+    size_t signed_data_size = 0;
+    PEP_STATUS status = sign_only(session, data, data_size, identity->fpr, &signed_data, &signed_data_size);
+    free(signed_data);
+    return status;
+}
```

Comment:
What is the point of this method? Just check if signing is possible?

Reply:
Yes, exactly, it's sort of a shortcut for checking if a key has a passphrase set.

While this passes all the tests that I have so far thrown at it, we can now do the same with more detail in the backend.

@ip-plank-security Shall we move the passphrase check into the backend? In this ticket or in a follow-up?

Reply:
@planckdz ok works for me